trio/_util.py

```diff
@@ -325,39 +331,39 @@ class SomeClass(metaclass=Final):
     ------
     - TypeError if a subclass is created
     """
+    # Override the method blindly. We're always going to raise, so it doesn't
+    # matter what the original did (if anything).
+    decorated.__init_subclass__ = classmethod(_init_final_cls)  # type: ignore[assignment]
+    # Python 3.11+ sets this attr, do it on all versions so this being present can be tested.
+    decorated.__final__ = True  # type: ignore[attr-defined]
```

Comment:
We could just call `typing.final` in case it ever grows new functionality.

Reply:
... Or actually that won't help with testing things. Uh.... I mean we can check `__init_subclass__` maybe?

Reply:
Definitely should call `typing.final` also, in case it does have additional runtime effects. Checking `__init_subclass__` feels a bit iffy since it's sorta leaking implementation details. Though these are tests. I could try actually subclassing every class, see how slow that is :P.

Reply:
We could unconditionally set `__final__` and call `typing.final` too ig